### PR TITLE
[EMBR-5419] Integrate native test flow for custom export config

### PR DESCRIPTION
**Pull request by facostaembrace**
## Goal

- Adding unit tests for otlp component
- Using local package instead of duplicating source files
- TBD: improve/add assertions

### CODE REVIEW

File: packages/react-native-otlp/test-project/android/app/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+dependencies {
+    implementation(libs.androidx.core.ktx)
+    implementation(libs.androidx.appcompat)
+    implementation(libs.material)
+
+    // react native 0.75.1
+    implementation("com.facebook.react:react-android:0.75.1")
+
+    // using package locally
+    implementation(project(":react-native-otlp"))
```
Comment (facostaembrace) [inline packages/react-native-otlp/test-project/android/app/build.gradle.kts:55]: @jpmunz please, have a look at this config. this is how I'm connecting the module into the `test-project` without copying/pasting the source files

File: packages/react-native-otlp/test-project/android/settings.gradle [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+dependencyResolutionManagement {
+    repositoriesMode.set(RepositoriesMode.PREFER_PROJECT)
+    repositories {
+        google()
+        mavenCentral()
+    }
+}
+
+// using local package
+include ':react-native-otlp'
```
Comment (facostaembrace) [inline packages/react-native-otlp/test-project/android/settings.gradle:39]: @jpmunz this (and next line) is also part of what we need to connect the module into the test-project without copying/pasting src files

File: packages/react-native-otlp/test-project/android/settings.gradle [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            content {
+                includeGroupByRegex("com\\.android.*")
+                includeGroupByRegex("com\\.google.*")
+                includeGroupByRegex("androidx.*")
+            }
+        }
+        gradlePluginPortal()
+    }
+
+    // including external React Native plugin (from root of monorepo)
```
Comment (jpmunz) [inline packages/react-native-otlp/test-project/android/settings.gradle:15]: based on the path this is just taking it from `test-project/node_modules` and not the root of the monorepo right?

File: packages/react-native-otlp/test-project/android/settings.gradle (hunk 2 above)
Comment (jpmunz) [reply packages/react-native-otlp/test-project/android/settings.gradle:39]: oh nice! Ya this is definitely a cleaner approach than the copying

File: packages/react-native-otlp/test-project/android/app/build.gradle.kts (hunk 1 above)
Comment (jpmunz) [reply packages/react-native-otlp/test-project/android/app/build.gradle.kts:55]: yep this is great, much better approach

**Review by jpmunz — APPROVED** [review]

File: packages/react-native-otlp/README.md [hunk 4]
```diff
@@ -1,8 +1,11 @@
-## React Native OTLP
+## Introduction
```
[the remark this answers is not in the document]
Comment (facostaembrace) [reply packages/react-native-otlp/README.md:1]: makes sense. following the other packages I'll update this one with the title `React Native Embrace - OTLP`

File: packages/react-native-otlp/android/settings.gradle
[the remark this answers is not in the document]
Comment (facostaembrace) [reply packages/react-native-otlp/android/settings.gradle:1]: will confirm, but ideally yeah I don't think `settings.gradle` should be packed. I'm using that file for adding the react native dependency + some local configuration for development only

**Pull request merged**